### PR TITLE
Validation Enhancements

### PR DESCRIPTION
**Pull request by Rangababu-R**
1. Got rid of regex entirely.
2. Added item validation of the list
3. addressed validation gaps.
#42 #44 #43 #47

### CODE REVIEW

File: openapiart/common.py [hunk 1]
```diff
@@ -187,101 +187,118 @@ def _decode(self, dict_object):
...
+        if value is None or not isinstance(value, str):
+            return False
+        return all([
+            True if int(bin) == 0 or int(bin) == 1 else False
+            for bin in value
+        ])
+    
+    def types_validation(self, value, type_, err_msg, itemtype=None):
+        type_map = {
+            int: "integer", str: "string",
```
Comment (alakendu) [inline openapiart/common.py:279]: list issue still present (https://github.com/open-traffic-generator/openapiart/issues/43)
I am getting error at the time of run this 
```python
    d = cfg.devices.device(name='d', container_name=p1.name)[-1]
    eth = d.ethernet
    eth.name = 'e'
    eth.mac = ['00:01:00:00:00:01']
```
```python
>           raise TypeError(err_msg)
E           TypeError: Invalid ['00:01:00:00:00:01'] format, expected mac at <class 'snappi.snappi.DeviceEthernet'>
```

File: openapiart/common.py [hunk 2]
```diff
@@ -187,101 +187,118 @@ def _decode(self, dict_object):
...
+            return all([0 <= int(oct) <= 255 for oct in ip.split(".", 3)])
         except Exception:
             return False
 
-    def _validate_ipv6(self, ip):
-        if ip is None:
-            return False
-        if self._IPV6_REP1.match(ip) or self._IPV6_REP2.match(ip):
+    def validate_ipv6(self, ip):
+        if ip is None or not isinstance(ip, str):
```
Comment (alakendu) [inline openapiart/common.py:218]: Not showing any error for trailing whitespaces
```python
ip.gateway = '2a00:1450:f013:c03:0:0:0:1 '
```

**Comment by ajbalogh** [comment]
Please check the format validation cases in test_formats.py that are named test_formats_ipv4_to_be_removed and test_formats_integer_to_be_removed that demonstrate values that should not be passing.

**Comment by Rangababu-R** [comment]
> 
> 
> Please check the format validation cases in test_formats.py that are named test_formats_ipv4_to_be_removed and test_formats_integer_to_be_removed that demonstrate values that should not be passing.

yes @ajbalogh I had added the fix for it and modified the testcase. its in my local. seeing an issue in snappi with choice. will fix and update the branch

File: openapiart/common.py (hunk 1 above)
Comment (Rangababu-R) [reply openapiart/common.py:279]: I think you cannot pass list to eth.mac right?

File: openapiart/common.py (hunk 2 above)
Comment (Rangababu-R) [reply openapiart/common.py:218]: I feel trailing spaces can be striped, we may need to test with illegal char or space with in the address. will check that part

**Review by ajbalogh — APPROVED** [review]

File: openapiart/tests/server.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+from flask import Flask, request, Response
```
Comment (alakendu) [inline openapiart/tests/server.py:1]: Include "flask" in pipeline. Otherwise build will fail

**Pull request merged**